poetry.lock

```diff
@@ -173,7 +173,7 @@ python-versions = "*"
 
 [[package]]
 name = "dulwich"
-version = "0.20.42"
+version = "0.20.44"
```

Comment:
If we're now relying on the behaviour of dulwich >= 0.20.44: then pyproject.toml ought to _require_ that dulwich, rather than continuing to require 0.20.35 and just trusting that a suitably recent version will be found

Reply:
Fixed.